bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/BagVerifier.scala

```diff
@@ -16,95 +16,100 @@ import uk.ac.wellcome.storage.{Location, Prefix}
 
 import scala.util.Try
 
-trait BagVerifier[BagLocation <: Location, BagPrefix <: Prefix[BagLocation]]
-    extends Logging
+sealed trait BagVerifyContext[BagLocation <: Location, BagPrefix <: Prefix[
+  BagLocation
+]] {
+  val root: BagPrefix
+}
```

Comment:
Uhm that doesn't seem to work. When I do this and the try to instantiate one, it says that it doesn't match the type boundaries. I think that's because `Prefix` is not covariant. I think I'll leave it as is unless you have strong opinions

Reply:
Sure; it'd be a nice cleanup if it works but not essential.